### PR TITLE
feat(react-native): Add theming for federated buttons

### PR DESCRIPTION
**Pull request by ioanabrooks**
#### Description of changes

Add style props and theming to federated buttons

| Before  | After |
| ------------- | ------------- |
| Simulator Screen Shot - iPhone 13 - 2023-06-21 at 17 59 03  | Simulator Screen Shot - iPhone 13 - 2023-07-19 at 18 13 52  |

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by dbanksdesign — APPROVED** [review]
LGTM!

File: packages/react-native/src/Authenticator/common/FederatedProviderButtons/types.ts [hunk 1]
```diff
@@ -1,16 +1,35 @@
...
+   * @description
+   * Styles for the divider text
+   */
+  textStyle?: StyleProp<TextStyle>;
+
+  /*
+   * @description
+   * Styles for the divider line
+   */
+  dividerStyle?: StyleProp<ViewStyle>;
```
Comment (calebpollman) [inline packages/react-native/src/Authenticator/common/FederatedProviderButtons/types.ts:27]: The [`Divider`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react-native/src/primitives/Divider/types.ts#L3) itself exposes `labelStyle` and `lineStyle` as well as a `style` prop that us applied to the containing `View`. Any reason not to extend from the existing types and follow the naming patterns?

File: packages/react-native/src/Authenticator/common/FederatedProviderButtons/types.ts [hunk 2]
```diff
@@ -1,16 +1,35 @@
...
-// TODO extend from ViewProps, add button style related style props
 export interface FederatedProviderButtonsProps
   extends Pick<
-    AuthenticatorMachineContext,
-    'route' | 'socialProviders' | 'toFederatedSignIn'
-  > {}
+      AuthenticatorMachineContext,
+      'route' | 'socialProviders' | 'toFederatedSignIn'
+    >,
+    ViewProps {
```
Comment (calebpollman) [inline packages/react-native/src/Authenticator/common/FederatedProviderButtons/types.ts:10]: Where are these additional props being applied?

File: packages/react-native/src/Authenticator/common/FederatedProviderButtons/types.ts [hunk 3]
```diff
@@ -1,16 +1,35 @@
...
+  /*
+   * @description
+   * Styles for the buttons
+   */
+  buttonStyle?: StyleProp<ViewStyle>;
```
Comment (calebpollman) [inline packages/react-native/src/Authenticator/common/FederatedProviderButtons/types.ts:15]: Think we should be extending from our `ButtonProps` here?

File: packages/react-native/src/Authenticator/common/FederatedProviderButtons/styles.ts [hunk 4]
```diff
@@ -1,7 +1,26 @@
...
+
+  return StyleSheet.create({
+    button: {
+      marginVertical: space.xs,
+    },
+    container: {
+      paddingHorizontal: space.small,
+      paddingVertical: space.xxs,
+    },
+    divider: {},
```
Comment (calebpollman) [inline packages/react-native/src/Authenticator/common/FederatedProviderButtons/styles.ts:20]: Why an empty object here?

**Pull request closed without merging**